### PR TITLE
🐛 Use product name instead of module name when patching XCFrameworks

### PR DESCRIPTION
**Pull request by PRESIDENT810**
### Description

When patching support files, we should use just product.name to replace XCFrameworks paths. This is because module name is used for importing modules in swift, which doesn't allow special characters like '-', so for modulemap search path we should use module name; however for header search path and library search path, special characters are allowed. Therefore when we have targets whose name has characters not supported in Swift, such as "foo-bar", using product.moduleName makes the regex pattern for patching XCFrameworks be `"${PODS_XCFRAMEWORKS_BUILD_DIR}/foo_bar"`, which causes search path `"${PODS_XCFRAMEWORKS_BUILD_DIR}/foo-bar"` cannot be substituted because regex pattern doesn't match.

For example:
image
image
For patching modulemap path, we should use `product.moduleName` which doesn't include special characters like '-', but for patching XCFramework, we should use `product.name`

### Checklist (I have ...)
- [x] 🧐 Followed the code style of the rest of the project
- [ ] 📖 Updated the documentation if necessary
- [x] 👨🏻‍🔧 Added at least one test that validates that my change is working, if appropriate
- [x] 👮🏻‍♂️ Run `make lint` and fixed all warnings
- [x] ✅ Run `make test` and fixed all tests

❤️ Thanks for contributing to the 🏈 Rugby!

### CODE REVIEW

File: Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift [hunk 1]
```diff
@@ -39,18 +39,19 @@ final class SupportFilesPatcher {
...
             let moduleMap = "\(moduleName).modulemap"
             result.append(("\(PODS_CONFIGURATION_BUILD_DIR)\(parentFolderName)/\(moduleMap)",
                            replacement: "\(binaryFolderPath)/\(moduleMap)"))
 
             if product.type == .staticLibrary {
                 result.append((
-                    "\(PODS_XCFRAMEWORKS_BUILD_DIR)\(moduleName)/\(HEADERS)",
-                    replacement: "\(binaryFolderPath)/\(moduleName)/\(HEADERS)"
+                    "\(PODS_XCFRAMEWORKS_BUILD_DIR)\(name)/\(HEADERS)",
+                    replacement: "\(binaryFolderPath)/\(name)/\(HEADERS)"
```
Comment (swiftyfinch) [inline Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift:50]: Let's use `product.name` directly in `"\(PODS_XCFRAMEWORKS_BUILD_DIR)\(product.name)/\(HEADERS)"`.
It looks like the `name` is redundant variable.

```suggestion
                    "\(PODS_XCFRAMEWORKS_BUILD_DIR)\(product.name)/\(HEADERS)",
                    replacement: "\(binaryFolderPath)/\(product.name)/\(HEADERS)"
```

File: Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift [hunk 2]
```diff
@@ -39,18 +39,19 @@ final class SupportFilesPatcher {
...
                 ))
                 result.append((
-                    "\(PODS_XCFRAMEWORKS_BUILD_DIR)\(moduleName)",
-                    replacement: "\(binaryFolderPath)/\(moduleName)"
+                    "\(PODS_XCFRAMEWORKS_BUILD_DIR)\(name)",
+                    replacement: "\(binaryFolderPath)/\(name)"
```
Comment (swiftyfinch) [inline Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift:54]: And the same here.

```suggestion
                    "\(PODS_XCFRAMEWORKS_BUILD_DIR)\(product.name)",
                    replacement: "\(binaryFolderPath)/\(product.name)"
```

File: Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift [hunk 3]
```diff
@@ -39,18 +39,19 @@ final class SupportFilesPatcher {
                            replacement: binaryFolderPath))
 
             guard let moduleName = product.moduleName else { return }
+            let name = product.name
```
Comment (swiftyfinch) [inline Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift:42]: And after that we can remove this line.
```suggestion
```

**Review by swiftyfinch — CHANGES_REQUESTED** [review]

File: Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift (hunk 1 above)
Comment (PRESIDENT810) [reply Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift]: Fixed

File: Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift (hunk 2 above)
Comment (PRESIDENT810) [reply Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift]: Fixed

File: Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift (hunk 3 above)
Comment (PRESIDENT810) [reply Sources/RugbyFoundation/Core/Use/SupportFilesPatcher.swift]: Fixed

**Comment by PRESIDENT810** [comment]
@swiftyfinch Ah sorry, wasn't checking GitHub notifications on weekends... I fixed them and now it should be OK.

**Review by swiftyfinch — APPROVED** [review]

**Comment by swiftyfinch** [comment]
@PRESIDENT810 Thank you for your contribution! 🚀

**Pull request merged**